lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * @license
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaseApi } from './base.api';
+import { throwIfNotDefined } from '../../../assert';
+import { ContentPagingQuery } from '../../content-rest-api';
+import { HoldPaging } from '../model/holdPaging';
+
+/**
+ * Legal Holds service.
+ *
+ * @module LegalHoldApi
+ */
+export class LegalHoldApi extends BaseApi {
+    /**
+     * List of legal holds
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param options Optional parameters
+     * @returns Promise<HoldPaging>
+     */
+    getHolds(filePlanId: string = '-filePlan-', options?: ContentPagingQuery): Promise<HoldPaging> {
```

Comment:
no need to set type to filePlanId when you assign value inline because ts can detect type based on value

Reply:
done